Anime-App/Feature/Anime/View/ViewControllers/AnimeViewController.swift

```diff
@@ -117,58 +123,91 @@ extension AnimeViewController {
         createSearchBar()
         view.addSubview(tableView)
         self.tableView.addSubview(self.refreshControl)
+        
+        customLoadingView.center = self.view.center
+        self.view.addSubview(customLoadingView)
+        customLoadingView.bringSubviewToFront(self.view)
+        
         spinner.center = self.view.center
         self.view.addSubview(spinner)
         spinner.bringSubviewToFront(self.view)
-
+        
         NSLayoutConstraint.activate([
+            
+            customLoadingView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            customLoadingView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            
             spinner.centerXAnchor.constraint(equalTo: view.centerXAnchor),
-            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            spinner.centerYAnchor.constraint(equalTo: view.centerYAnchor, constant: 50),
+            
             tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
             tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor, constant: 16),
             tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor, constant: -16),
             tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
         ])
     }
     
-    func fetchAnime() {
-        self.isFetching = true
+    func resetAnime() {
+        animeViewModel.resetAnime { res in
+            switch res {
+            case .success:
+                self.tableView.reloadData()
+            case .failure:
+                print("Failed to reset")
+            }
+        }
+    }
+    
+    func startPagination() {
+        self.isPaginating = true
         self.spinner.startAnimating()
+        spinner.isHidden = false
         animeViewModel.fetchAnime(page: page)  { [weak self] res in
             switch res {
             case .success:
                 self?.tableView.reloadData()
-             case .failure:
+            case .failure:
                 print("no more data")
-             }
+            }
             self?.spinner.stopAnimating()
-            self?.isFetching = false
+            self?.spinner.isHidden = true
+            self?.isPaginating = true
         }
     }
     
-    func resetAnime() {
-        animeViewModel.resetAnime { res in
-            switch res {
-            case .success:
-                self.tableView.reloadData()
-            case .failure:
-                print("Failed to reset")
+    
+    func fetchAnime() {
```

Comment:
When you fetch anime the only property you want to change in here is isFetching, I'm not too sure why you'd want to set/change the pagination since this can cause side effects (behaviour you don't expect) i wouldn't be changing the pagination property in here you should only be changing the property that this function needs to use which looks like isFetching